AuroraAssetEditor/Classes/XboxAssetDownloader.cs

```diff
@@ -52,25 +53,21 @@ internal class XboxAssetDownloader {
             try {
             var ret = new List<XboxLocale>();
             var tmp = new List<string>();
-            var wc = new WebClient();
-            var data = Encoding.UTF8.GetString(wc.DownloadData("http://www.xbox.com/Shell/ChangeLocale")).Split('>');
-            for(var i = 0; i < data.Length; i++) {
-                if(!data[i].ToLower().Contains("?source=lp"))
-                    continue;
-                var index = data[i].ToLower().IndexOf("\"pid\":", StringComparison.Ordinal) + 7;
-                var id = data[i].Substring(index);
-                index = id.IndexOf('"');
-                if(index <= 0)
-                    continue;
-                id = id.Substring(0, index);
-                if(tmp.Contains(id))
-                    continue;
-                var name = data[i + 1];
-                name = name.Substring(0, name.IndexOf("</a", StringComparison.Ordinal));
-                name = name.Trim('\r','\n',' ');
-                name = HttpUtility.HtmlDecode(name);
-                ret.Add(new XboxLocale(id, name));
-                tmp.Add(id);
+            var wc = new WebClient();
+            // Enable TLS 1.2 (3072) security protocol
+            // https://docs.microsoft.com/en-us/dotnet/api/system.net.securityprotocoltype?view=net-6.0
+            System.Net.ServicePointManager.SecurityProtocol |= (SecurityProtocolType)(3072);
+            string text = Encoding.UTF8.GetString(wc.DownloadData("https://www.xbox.com/en-US/Shell/ChangeLocale"));
+            string search = "<a.*?href=\"(.+?)\".*?>(.+?)</a>";
+            MatchCollection matches = Regex.Matches(text, search);
+            foreach (Match m in matches) {
+                if (m.Groups.Count >= 2 && m.Groups[1].ToString().EndsWith("/?source=lp")) {
+                    var id = m.Groups[1].ToString().ToLower();
+                    var index = id.IndexOf("/?source=lp") - 1;
+                    id = id.Substring(1, index);
+                    var name = m.Groups[2].ToString();

```

Comment:
Could you add back support for decoding this data to a normal state as the original code does?

Reply:
Updated the `fix locale parser` commit to have the following changes:

- Revert `text` variable name to `data`
- Update regex to only capture `id` from the href
- Trim `id` and `name`
- HTML decode `name`